src/crimson/os/seastore/transaction.h

```diff
@@ -157,6 +157,10 @@ class Transaction {
       delayed_temp_offset += ref->get_length();
       delayed_alloc_list.emplace_back(ref->cast<LogicalCachedExtent>());
       fresh_block_stats.increment(ref->get_length());
+    } else if (ref->get_paddr().get_addr_type() == paddr_types_t::RANDOM_BLOCK) {
+      assert(ref->is_logical());
+      delayed_alloc_list.emplace_back(ref->cast<LogicalCachedExtent>());
+      fresh_block_stats.increment(ref->get_length());
```

Comment:
Transaction/TransactionManager doesn't need to be specific to backend/paddr types.

Generally, there will be the following 3 kinds of fresh extents possible here after this PR:
1. Inline blocks that will be allocated&written later with the journal record.
2. Delayed alloc blocks that will be allocated&written later (maybe inline or ool) in `EPM::delayed_alloc_or_ool_write()`.
3. Pending ool blocks that is already allocated but needs to be written later.

So, we might need to add another block list to `Transaction` for the new case 3, probably named `allocated_ool_block_list`, so that the logic here becomes:
```c++
} else if (ref->get_paddr().is_absolute()) {
  allocated_ool_block_list.emplace_back(ref);
  fresh_block_stats.increment(ref->get_length());
} else { ...
```

Then, in the transaction phase of `tref.get_handle().enter(write_pipeline.ool_writes)`, we can add another path so that the extents in `allocated_ool_block_list` will be written with `EPM::proceed_ool_write()` and then be added to `Transaction::ool_block_list` for the later transaction phases.

Note the above logic needs proper further refactors because `EPM::proceed_ool_write()` is different from `EPM::delayed_alloc_or_ool_write()` (probably better renamed as `EPM::delayed_allocate_and_write()` now) that the already allocated extents don't need to update the lba tree mappings. And the both paths can be called in parallel to achieve parallel writes.

I think we might need some review iterations to implement this properly.